monkey/infection_monkey/exploit/HostExploiter.py

```diff
@@ -101,6 +104,14 @@ def pre_exploit(self):
         )
         self.set_start_time()
 
+    def is_interrupted(self):
+        # This method should be refactored to raise an exception to reduce duplication in the
+        # "if is_interrupted: return self.exploitation_results"
+        # Ideally the user should only do "check_for_interrupt()"
+        if self.interrupt.is_set():
+            logger.info("Exploiter has been interrupted by a stop signal from the Island")
```

Comment:
Consider setting `self.exploit_result.error_message` so the user can know that exploitation failed due to an interrupt.